pages/blog/[uid].js

```diff
@@ -42,12 +45,12 @@ const Post = ({ post }) => {
   return null;
 };
 
-export async function getStaticProps({ params, preview = null, previewData = {} }) {
-  const { ref } = previewData
+export async function getStaticProps({ params, previewData }) {
+  const ref = previewData ? previewData.ref : null
   const post = await Client().getByUID("post", params.uid, ref ? { ref } : null) || {}

```

Comment:
For the sake of making this easier to read, I would move the `ref ? { ref } : null` logic to its own line above this and assign it in a const named `options`.